File: frontend-application/src/app/model/Note.ts
import {Category} from "./Category";
import {Status} from "./Status";

export class Note {
  id: string;
  name: string;
  description: string;
  category: Category;
  favorite: boolean;
  created: Date;
  updated: Date;
  status: Status;
  userId: string;

  constructor(id: string, name: string, description: string, category: Category, favorite: boolean, created: Date, updated: Date, status: Status, userId: string) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.category = category;
    this.favorite = favorite;
    this.created = created;
    this.updated = updated;
    this.status = status;
    this.userId = userId;
  }
}

File: backend-application/src/main/java/com/localnotes/entity/Note.java
package com.localnotes.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "notes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Note extends BaseEntity {

    @Column(name = "note_name")
    private String name;

    @Column(name = "description")
    private String description;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.DETACH, CascadeType.REFRESH, CascadeType.MERGE})
    @JoinColumn(name = "cat_id", referencedColumnName = "id")
    private Category category;

    @Column(name = "user_id")
    private String userId;

    @Column(name = "favorite")
    private boolean isFavorite;
}
